pkg/report/datafeeds/types.go

```diff
@@ -2,53 +2,95 @@ package datafeeds
 
 import (
 	"fmt"
+	"math/big"
 
 	"github.com/ethereum/go-ethereum/accounts/abi"
 )
 
 // This is ABI encoding - abi: "(bytes32 FeedID, bytes RawReport)[] Reports" (set in workflow)
 // Encoded with: https://github.com/smartcontractkit/chainlink/blob/develop/core/services/relay/evm/cap_encoder.go
 type FeedReport struct {
-	FeedID [32]byte
-	Data   []byte
+	FeedID    [32]byte
+	Timestamp uint32
+	Price     *big.Int // *big.Int is used because go-ethereum converts large uints to *big.Int.
 }
 
-type Reports = []FeedReport
+type CCIPFeedReport struct {
+	FeedID    [32]byte
+	Price     *big.Int
+	Timestamp uint32
+}
```

Comment:
It's not clear how this changed so drastically. Why did it change so drastically?

Before, we had a Mercury report (underlying) wrapped in a FeedReport (root), now there's no underlying report anymore?

Reply:
That's correct, there is no underlying report for these types. Keep in mind these or for EVM. On Aptos there would a procesor that uses the structure you are describing. I left that out of scope for the current set of PRs though since we would not be using that any time soon on Aptos.